pkg/streamingpromql/operators/functions/rate_increase.go

```diff
@@ -122,15 +132,20 @@ func histogramRate(isRate bool, hCount int, hHead []promql.HPoint, hTail []promq
 		return nil
 	}
 
-	err = accumulate(hHead)
-	if err != nil {
-		return nil, err
+	if isCounter {
```

Comment:
Should `previousValue` and `accumulate` move inside this `if`? They seem to only be used here.

Reply:
This comment is not relevant anymore.